android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapInterfaceController.kt

```diff
@@ -253,6 +284,34 @@ class MapInterfaceController(
     }
   }
 
+  @OptIn(MapboxExperimental::class, MapboxDelicateApi::class)
+  override fun setFeatureStateForFeaturesetDescriptor(
+    featureset: FeaturesetDescriptor,
+    featureId: FeaturesetFeatureId,
+    state: Map<String, Any?>,
+    callback: (Result<Unit>) -> Unit
+  ) {
+    mapboxMap.setFeatureState(
+      featureset.toTypedFeaturesetDescriptor() as TypedFeaturesetDescriptor<FeatureState, com.mapbox.maps.interactions.FeaturesetFeature<FeatureState>>,
+      featureId.toFeaturesetFeatureId(),
+      state.toFeatureState()
+    ) { callback(Result.success(Unit)) }
+  }
+
+  @OptIn(MapboxExperimental::class, MapboxDelicateApi::class)
+  override fun setFeatureStateForFeaturesetFeature(
+    feature: FeaturesetFeature,
+    state: Map<String, Any?>,
+    callback: (Result<Unit>) -> Unit
+  ) {
+    val id = feature.id?.toFeaturesetFeatureId() ?: return
```

Comment:
We can't return without invoking the callback - dart will hang invoking async execution context until reply is delivered.

Reply:
Good catch -- I've updated all of these